src/authentication/git-auth-scheme.ts

```diff
@@ -0,0 +1,172 @@
+import * as auth0 from 'auth0-js';
+import * as Boom from 'boom';
+import { decode, verify } from 'jsonwebtoken';
+import * as jwksRsa from 'jwks-rsa';
+import fetch from 'node-fetch';
+import * as util from 'util';
+
+import * as Hapi from '../server/hapi';
+import getResponseJson from '../shared/get-response-json';
+import * as logger from '../shared/logger';
+import { sanitizeSubClaim } from './authentication-hapi-plugin';
+import { AccessToken } from './types';
+
+interface Auth0LoginResponse {
+  accessToken: string;
+  expisresIn: number;
+}
+
+export class GitAuthScheme {
+  private readonly _getSigningKey: (arg1: string) => Promise<jwksRsa.Key>;
```

Comment:
Instead of having these underscore functions, I'd prefer it if the name describes what it exactly is. The underscore gives no indication into why the underscore is there or why it's different from a non-underscore version.